modules/api/src/main/java/org/apache/ignite/network/IgniteCluster.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Ignite cluster. Provides access to cluster nodes and the local node.
+ */
+public interface IgniteCluster {
+    /**
+     * Gets the cluster nodes.
+     *
+     * @return Collection of cluster nodes.
+     */
+    Collection<ClusterNode> nodes();
```

Comment:
The Javadoc looks vague to me. The current implementation just returns `TopologyService.allMembers()` that assumes that the returned collection may contain nodes from the physical topology that are not part of the logical topology. Should we filter out such nodes? IMHO, we need to add something to clarify that at least.

Reply:
Good catch! Currently, this API is mostly used together with Compute to submit jobs.
And nodes that are not a part of the logical topology can't run compute jobs, so current implementation will result in an error.

I'll change it to return only the members of the logical topology.

Reply:
Perhaps, we need a bunch of tests to cover such a scenario (in a separate ticket of course).

Reply:
Fixed the code to return logical topology members only.

Ticket for additional tests: https://issues.apache.org/jira/browse/IGNITE-25662